crates/sparrow-merge/src/gather.rs

```diff
@@ -46,6 +59,10 @@ impl Gatherer {
         self.pending[index].add_batch(batch);
 
         if let Some(mut entry) = self.active.pop() {
+            // Expect `blocking_input` to be called first, ensuring we're working
+            // on the current blocking input
+            assert_eq!(entry.index, index);
```

Comment:
Maybe add a message to the assertion?

`assert_e!(entry.index, index, "Expected next batch from blocking input {} but was {index}", entry.index)`